docs/source/topics/backends/snmp.rst

```diff
@@ -0,0 +1,149 @@
+
+************
+SNMP backend
+************
+
+SNMP
+====
+
+SNMP (Simple Network Management Protocol) is a network protocol very useful for retrieving info from a device.
+All the information is retrieved by using codes called MIBs. All MIBs have a tree-like structure, every main information is the root and by adding more detail to the info
+the tree gains more depth.
+Obviously, by getting the smallest MIB which is "1" or simply " . " one can get all the tree.
+
+The base SNMP backend contains the following methods (some internal methods are not documented and are subject to change in the future):
+
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_dict**  | Returns a dict containing monitoring information depending on the type of the device.                                                    |
+|              | It follows the `NetJSON Devicemonitoring <https://netjson.org/rfc.html#name-devicemonitoring>`_ spec                                     |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_json**  | Calls the `to_dict` method and returns a JSON string of the dict                                                                         |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **validate** | Checks if connection with the device is working and raises `NetengineError` in case something is wrong                                   |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
```

Comment:
I suggest to do the same here and drop the table syntax which is painful to maintain